op-chain-ops/genesis/config.go

```diff
@@ -623,6 +624,20 @@ func (d *DeployConfig) RollupConfig(l1StartBlock *types.Block, l2GenesisBlockHas
 		}
 	}
 
+	chConfig, ok := superchain.OPChains[d.L2ChainID]
+	if !ok {
+		return nil, fmt.Errorf("unknown chain ID: %d", d.L2ChainID)
+	}
+
+	superChain, ok := superchain.Superchains[chConfig.Superchain]
```

Comment:
I don't think its good to error here, then this cannot be used to create new rollups outside of what is in the superchain registry

Reply:
Good call. Fixed here: https://github.com/ethereum-optimism/optimism/pull/11018/commits/1e72e3d3e12b9156c00d79b5c17c71a279591c52

Do you see any issues with setting the `ProtocolVersionsAddress` here rather than further downstream? I am trying to fix a sanity check test failure found in [this superchain-registry PR](https://github.com/ethereum-optimism/superchain-registry/pull/316) where the `rollupConfig` Unmarshaled from `rollup.json` does not match the one produced by op-node `LoadOPStackRollupConfig`. Another option would be to just exclude that field from the sanity check or make some other sort of special case/exception

Reply:
I think its fine to include it here but I would prefer to pass it as an argument to this function so that its more flexible. We can put the code for optionally fetching the address [here](https://github.com/ethereum-optimism/optimism/blob/3d745c58146b7528dbeaa1561b2935a8b0eefc2c/op-node/cmd/genesis/cmd.go#L229) via a new cli flag and if its not set via cli then fallback to trying to get it from the registry without erroring and just pass thru `address(0)`

Reply:
This will enable us to easily create devnets or new testnets that are not in the registry as well as external teams that use our software to do whatever in the most flexible way

Reply:
Thought about this some more. It seems, based on your comments, that populating the `ProtocolVersionAddress` field at this point can alter functionality. I don't think it makes sense to change functionality just for the sake of satisfying a sanity check test in the superchain-registry. I will close this PR.

Instead, I altered the [superchain-registry PR](https://github.com/ethereum-optimism/superchain-registry/pull/316/commits/ffdd7e151ae4a820f26382e74fba408dba22a330) to just exclude the `ProtocoalVersionAddress` field from the `add-chain check-rollup-config` subcommand.